avocado/utils/config_change_validation.py

```diff
@@ -0,0 +1,205 @@
+
+        if len(compared_file_paths) > i:
+            command = ("diff -U 0 -b " + compared_file_paths[i]
+                       + " " + file_paths[i])
+        else:
+            command = "diff -U 0 -b " + temp_file_path + " " + file_paths[i]
+
+        output = process.run(command, ignore_status=True).stdout_text
+        lines = output.split('\n')
+        changes[file_paths[i]] = parse_unified_diff_output(lines)
+    return changes
+
+
+def assert_config_change_dict(actual_result, expected_result):
+    """
+    Calculates unexpected line changes.
+
+    :param actual_result: actual added and removed lines
+    :type actual_result: (file_path, ([added_line, ...], [removed_line, ...])) dict
+    :param expected_result: expected added and removed lines
+    :type expected_result: (file_path, ([added_line, ...], [removed_line, ...])) dict
+    :returns: detected differences as groups of lines
+    :rtype: (file_path, (unexpected_adds, not_present_adds, unexpected_removes, not_present_removes)) dict
+    """
+    change_diffs = {}
+    for file_path, actual_changes in actual_result.items():
+        expected_changes = expected_result[file_path]
+
+        actual_adds = actual_changes[0]
+        actual_removes = actual_changes[1]
+        expected_adds = expected_changes[0]
+        expected_removes = expected_changes[1]
+
+        # Additional unexpected adds -- they should have been not added
+        unexpected_adds = sorted(set(actual_adds) - set(expected_adds))
+        # Not present expected adds -- they should have been added
+        not_present_adds = sorted(set(expected_adds) - set(actual_adds))
+        # Additional unexpected removes - they should have been not removed
+        unexpected_removes = sorted(set(actual_removes) - set(expected_removes))
+        # Not present expected removes - they should have been removed
+        not_present_removes = sorted(set(expected_removes) -
+                                     set(actual_removes))
+
+        change_diffs[file_path] = (unexpected_adds, not_present_adds,
+                                   unexpected_removes, not_present_removes)
+
+    return change_diffs
+
+
+def assert_config_change(actual_result, expected_result):
+    """
+    Condition wrapper of the upper method.
+
+    :param actual_result: actual added and removed lines
+    :type actual_result: (file_path, ([added_line, ...], [removed_line, ...])) dict
+    :param expected_result: expected added and removed lines
+    :type expected_result: (file_path, ([added_line, ...], [removed_line, ...])) dict
+    :returns: whether config changes were detected
+    :rtype: bool
+    """
+    change_diffs = assert_config_change_dict(actual_result, expected_result)
+    for file_change in change_diffs.values():
+        for line_change in file_change:
+            if len(line_change) != 0:
+                return False
+    return True
+
+
+def print_change_diffs(change_diffs):
```

Comment:
IMO, the name of this method gives the idea that text will be printed out on the screen (by `print()`, `logging`, etc).  So, I'd suggest you name it something like `create_diff_report()` or `create_human_diff_summary()`.

Reply:
Resolved on the next push.